WordPress/gutenberg
# Plugin: Remove PHP functions slated for removal in 5.2

This pull request seeks to remove functions deprecated as part of the Gutenberg 5.0 release, slated for removal with the upcoming version 5.2.

**Implementation notes:**

As noted at https://github.com/WordPress/gutenberg/issues/14012#issuecomment-467073907, a few functions were wrongly labeled as deprecated with 5.0 when in fact they were deprecated in the 5.1 release. Those have not been removed here, and their version deprecations have been updated to align to the deprecations document, where they are correctly noted as being removed in the following 5.3 release:

https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/backward-compatibility/deprecations.md

The script `bin/get-server-blocks.php` still included a reference to `gutenberg_prepare_blocks_for_js`. The deprecated warning was not logged because the file disables error reporting. This has been updated accordingly here. Separately, we should consider:

- Enable error reporting
- Run the fixtures regeneration, because its out of sync
- Find ways to avoid ever allowing (a) the fixtures to fall out of sync or (b) silently swallow errors

Related: https://github.com/WordPress/gutenberg/pull/13583#discussion_r259921366

**Testing instructions:**

Verify there are no regressions in the load and operation of the error.

Verify there are no warnings or errors logged (by PHP) when loading the editor (with and without meta boxes) or the reusable blocks list screen.

Ensure there are no remaining occurrences of deprecated functions to be removed in the 5.2 release.
